### PR TITLE
SPMI fixes for missing info.

### PR DESCRIPTION
**Pull request by sandreenko**
Improve SPMI experience when there are misses, they could be due to stale collection or due to local changes.
It is useful to separate Jit asserts from such misses when working on big changes that could introduce misses but should not introduce Jit asserts.

Link https://github.com/dotnet/runtime/issues/48356.
Before:
```
Running SuperPMI replay of D:\Sergey\git\runtime\artifacts\spmi\mch\af37688b-d4e5-4a41-a7ee-701728a470aa.windows.x64\tests.pmi.windows.x64.checked.mch
Compilation failures

Repro .mc files created for failures:
D:\Sergey\git\runtime\artifacts\spmi\repro.windows.x64.Checked.20\repro-102534.mc
***
D:\Sergey\git\runtime\artifacts\spmi\repro.windows.x64.Checked.20\repro-74002.mc
D:\Sergey\git\runtime\artifacts\spmi\repro.windows.x64.Checked.20\repro-80696.mc

To run a specific failure (replace JIT path and .mc filename as needed):

D:\Sergey\git\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\superpmi.exe  D:\Sergey\git\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\clrjit.dll D:\Sergey\git\runtime\artifacts\spmi\repro.windows.x64.Checked.20\xxxxx.mc
```

After:
```
Running SuperPMI replay of D:\Sergey\git\runtime\artifacts\spmi\mch\af37688b-d4e5-4a41-a7ee-701728a470aa.windows.x64\libraries.pmi.windows.x64.checked.mch
SPMI misses found
Running SuperPMI replay of D:\Sergey\git\runtime\artifacts\spmi\mch\af37688b-d4e5-4a41-a7ee-701728a470aa.windows.x64\tests.pmi.windows.x64.checked.mch
SPMI misses found

```

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @dotnet/jit-contrib

File: src/coreclr/scripts/superpmi.py [hunk 1]
```diff
@@ -1285,6 +1285,8 @@ def print_superpmi_failure_code(return_code, coreclr_args):
         logging.warning("Compilation failures")
     elif return_code == 2:
         logging.warning("Asm diffs found")
+    elif return_code == 3:
+        logging.warning("SPMI misses found")
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi.py:1289]: Can you please spell this out, e.g., "SuperPMI missing data encountered".

**Review by BruceForstall — CHANGES_REQUESTED** [review]

File: src/coreclr/ToolBox/superpmi/superpmi-shared/methodcontext.cpp [hunk 2]
```diff
@@ -2696,6 +2697,9 @@ CORINFO_CLASS_HANDLE MethodContext::repGetArgClass(CORINFO_SIG_INFO*       sig,
     key.scope                  = CastHandle(sig->scope);
     key.args                   = CastHandle(args);
 
+    AssertCodeMsg(GetArgClass != nullptr, EXCEPTIONCODE_MC,
```
[the remark this answers is not in the document]
Comment (sandreenko) [reply src/coreclr/ToolBox/superpmi/superpmi-shared/methodcontext.cpp]: sure, done

File: src/coreclr/scripts/superpmi.py (hunk 1 above)
Comment (sandreenko) [reply src/coreclr/scripts/superpmi.py]: done

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**